src/services/add-query-to-queue.ts

```diff
@@ -167,4 +181,67 @@ export default class AddQueryToQueue {
       await interaction.editReply(`u betcha, **${firstSong.title}** and ${newSongs.length - 1} other songs were added to the queue${extraMsg}`);
     }
   }
+
+  private async enrichWithSkipSegments(songs: SongMetadata[]) {
```

Comment:
might be a little cleaner to change the signature to
```suggestion
  private async enrichWithSkipSegments(song: SongMetadata) {
```
and have the caller worry about batching